preview/infrastructure/harvester/provider.tf

```diff
@@ -0,0 +1,37 @@
+terraform {
+
+  backend "gcs" {
+    bucket = "3f4745df-preview-tf-state"
+    prefix = "preview"
+  }
+
+  required_version = ">= 1.2"
+  required_providers {
+    harvester = {
+      source  = "harvester/harvester"
+      version = ">=0.5.1"
+    }
+    k8s = {
+      source  = "hashicorp/kubernetes"
+      version = ">= 2.0"
+    }
+  }
+}
+
+# https://registry.terraform.io/providers/harvester/harvester/latest/docs
+provider "harvester" {
+  alias      = "harvester"
+  kubeconfig = file(var.harvester_kube_path)
+}
+
+# https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs
+provider "k8s" {
+  alias       = "dev"
+  config_path = var.dev_kube_path
+}
+
+# https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs
+provider "k8s" {
+  alias       = "harvester"
+  config_path = var.harvester_kube_path
+}
```

Comment:
How come our pre-commit hook `end-of-file-fixer` didn't remove these missing newlines? ☺️

Reply:
Speaking of pre-commit-hooks it would be nice to add one for terraform fmt https://github.com/antonbabenko/pre-commit-terraform#3-add-configs-and-hooks

Reply:
Ah I commit with `--no-verify` by default and usually at the end I run the pre-commit manually to fix anything outstanding 😅  Weird that we don't run the pre-commit as a step in the CI so it fails in cases like this.

I'll add the `terraform fmt` in another PR with the rest of the TF.